catalyst/core/callbacks/wrapper.py

```diff
@@ -0,0 +1,361 @@
+        )
+    return filter_fn
+
+
+def _filter_fn_from_arg(filter_fn: Union[str, FILTER_FN]) -> FILTER_FN:
+    """Check if filter function from argumets
+    can be used with ``WrapperCallback``.
+
+    Args:
+        filter_fn (str or Callable): filter function to check
+
+    Raises:
+        ValueError: if ``filter_fn`` is a string and can not be
+            interpreted as python code then an error will be raised
+        ValueError: if passed not callable object then will be
+            raised an error
+        ValueError: will be raised error if filter function do not
+            have three arguments
+
+    Returns:
+        filter function which accepts 3 arguments - stage (str),
+        epoch (int), loader (str) and return ``True`` if
+        need to disable callback
+    """
+    if isinstance(filter_fn, str):
+        # lambda function from string
+        try:
+            filter_fn = eval(filter_fn)  # noqa: WPS400
+        except (ValueError, SyntaxError):
+            raise ValueError(
+                "'filter_fn' should be a valid "
+                "python lambda function with "
+                "three arguments - 'stage', 'epoch' and 'loader'!"
+            )
+    if not callable(filter_fn):
+        raise ValueError("'filter_fn' should be a callable!")
+    if filter_fn.__code__.co_argcount != 3:
+        raise ValueError(
+            "Filter function should have three arguments - "
+            "'stage', 'epoch' and 'loader'!"
+        )
+    return filter_fn
+
+
+class WrapperCallback(Callback):
+    """
+    Customize callback execution on different stages, loaders and epoch.
+
+    For example, if you don't want to compute loss on a validation
+    you can ignore ``CriterionCallback``, for notebook API
+    need to wrap callback:
+
+    .. code-block:: python
+
+        import torch
+        from torch.utils.data import DataLoader, TensorDataset
+        from catalyst.dl import (
+            SupervisedRunner, CriterionCallback, WrapperCallback
+        )
+
+        num_samples, num_features = 10_000, 10
+        X, y = torch.rand(num_samples, num_features), torch.rand(num_samples)
+        loader = DataLoader(TensorDataset(X, y), batch_size=32, num_workers=1)
+        loaders = {"train": loader, "valid": loader}
+
+        model = torch.nn.Linear(num_features, 1)
+        criterion = torch.nn.MSELoss()
+        optimizer = torch.optim.Adam(model.parameters())
+        scheduler = torch.optim.lr_scheduler.MultiStepLR(optimizer, [3, 6])
+
+        runner = SupervisedRunner()
+        runner.train(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            loaders=loaders,
+            logdir="./logdir",
+            num_epochs=8,
+            verbose=True,
+            callbacks=[
+                WrapperCallback(
+                    base_callback=CriterionCallback(),
+                    loaders=["valid"]
```

Comment:
🤔  based on provided example, should we split it's functionality to 
```
loaders=[] # loader names, where callback should be executed
ignore_loaders=[] # loader names, where we should ignore the callback
```
?

just wondering, how it would be more user-friendly

Reply:
btw, it also looks like this examples should not work because by default our metric - "loss" on "valid" loader, that is not computed here :)

Reply:
I think, we could easily use `dl.AccuracyCallback` instead